Objects/unicodeobject.c

```diff
@@ -13428,6 +13428,9 @@ PyUnicodeWriter_Create(Py_ssize_t length)
 
 void PyUnicodeWriter_Discard(PyUnicodeWriter *writer)
 {
+    if (writer == NULL) {
```

Comment:
This should be documented. Please do this in a separate PR.

Also, I think it is worth to try to optimize `PyUnicodeWriter_WriteUTF8()` for short ASCII strings.